src/foam/nanos/http/LiveScriptBundler.java

```diff
@@ -0,0 +1,353 @@
+        Set<Path> directories = new LinkedHashSet<>();
+        for (String foamName : paths) {
+          Path f = Paths.get(path_, currentFilesPath.getKey(), foamName);
+
+          // Add containing folder to a set for registering watchers
+          directories.add(f.getParent());
+
+          fileNames_.add(foamName);
+        }
+
+        // Register a separate thread to watch each directory.
+        // (this is necessary with WatchService)
+        for (Path d : directories) {
+          WatchService watcher = FileSystems.getDefault().newWatchService();
+          d.register(watcher,
+            new WatchEvent.Kind[]{ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY},
+            SensitivityWatchEventModifier.HIGH
+          );
+
+          // Find relative path from `src` folder to get foam path
+          Path relative = Paths.get(path_, currentFilesPath.getKey()).relativize(d.toAbsolutePath()).normalize();
+
+          Thread watcherThread = new Thread(new WatcherThread(
+            watcher,
+            d,
+            relative.toString(),
+            this::doRebuildJavascript
+          ));
+          watcherThread.start();
+        }
+      }
+    } catch (Throwable t) {
+      t.printStackTrace();
+      System.err.println("Failed to initialize filesystem watcher! :(");
+      System.exit(1);
+    }
+  }
+
+  private void doRebuildJavascript(String foamName, Path realPath) {
+    try {
+      System.out.println("Building javascript... (js)");
+
+      Process p = new ProcessBuilder(JS_BUILD_PATH).start();
+
+      BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()));
+      String line;
+      while ( (line=br.readLine()) != null ) {
+        System.out.println("js> " + line);
+      }
+
+      String contents = new String(Files.readAllBytes(Paths.get(FOAM_BIN_PATH)));
+      javascriptBuffer_  = contents;
+      System.out.println("[DONE] (js)");
+    } catch (IOException e) {
+      log_("ERROR", e.getMessage());
+    }
+  }
+
+  @Override
+  public void execute(X x) {
+    PrintWriter pw = x.get(PrintWriter.class);
+    HttpServletResponse r = x.get(HttpServletResponse.class);
+    r.setHeader("Content-Type", "application/javascript");
+    pw.println(javascriptBuffer_);
+    return;
+  }
+
+  // Proof-of-concept parser, to be replaced with a foam.lib.parse.Parser soon
+  private List<String> parseFilesjs(
+    BufferedReader reader
+  ) throws java.io.IOException {
+    // This is what we want to get
+    List<String> paths = new ArrayList<>();
+
+    // For one specific line in iso20022 files.js
+    boolean lineException = false;
+
+    for (
+      String line = reader.readLine() ;
+      line != null                    ;
+      line = reader.readLine()
+    ) {
+      line = line.trim();
+      if (line.length() < 1) continue;
+      if ( lineException ) {
+        line = "{" + line + "}";
+        lineException = false;
+      }
+      else if (line.charAt(0) != '{') continue;
```

Comment:
Please match the style of the 'if' and 'else if' parts, like this:

if ( lineException ) {
  line = "{" + line + "}";
  lineException = false;
} else if (line.charAt(0) != '{') {
  continue;
}

Reply:
Updated